macros/materializations/model.sql

```diff
@@ -54,18 +54,30 @@
     {%- set ml_config = config.get('ml_config', {}) -%}
     {%- set raw_labels = config.get('labels', {}) -%}
     {%- set sql_header = config.get('sql_header', none) -%}
+    {%- set has_sql = true -%}
+
+    {%- if ml_config.get('MODEL_TYPE', ml_config.get('model_type', '')).lower() == 'tensorflow' -%}
+        {%- set has_sql = false -%}
+    {%- endif -%}
 
     {{ sql_header if sql_header is not none }}
 
     create or replace model {{ relation }}
+
+    {% if ml_config.get("connection_name") %}
+        remote with connection `{{ ml_config.pop("connection_name") }}`
+        {% set has_sql = false %}
+    {% endif %}
+
     {{ dbt_ml.model_options(
         ml_config=ml_config,
         labels=raw_labels
     ) }}
-    {%- if ml_config.get('MODEL_TYPE', ml_config.get('model_type', '')).lower() != 'tensorflow' -%}
-    as (
-        {{ sql }}
-    );
+
+    {%- if has_sql -%}
```

Comment:
Would it perhaps be possible to automatically determine whether or not the model `has_sql`? Would it be sufficient to check whether or not the `sql`-variable contains only whitespace?

Reply:
You're absolutely right; the control flow can be made much more concise. I've updated the pull request removing the intermediary has_sql variable and put a condition directly asserting existence or lack thereof of sql as such: {%- if sql -%}
        as (
            {{ sql }}
        );
    {%- endif -%}